src/resourcemanager/resourcemanager.cpp

```diff
@@ -0,0 +1,490 @@
+/*
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <algorithm>
+#include <filesystem>
+#include <fstream>
+
+#include <utils/exception.hpp>
+#include <utils/json.hpp>
+
+#include "logger/logmodule.hpp"
+#include "resourcemanager.hpp"
+
+namespace aos::sm::resourcemanager {
+
+using common::utils::CaseInsensitiveObjectWrapper;
+using common::utils::GetArrayValue;
+
+/***********************************************************************************************************************
+ * Static
+ **********************************************************************************************************************/
+
+static std::vector<std::string> ParseStringArray(const CaseInsensitiveObjectWrapper& object, const std::string& key)
+{
+    return GetArrayValue<std::string>(
+        object, key, [](const Poco::Dynamic::Var& value) { return value.convert<std::string>(); });
+}
+
+static CaseInsensitiveObjectWrapper CreateCaseInsensitiveObjectWrapper(const Poco::Dynamic::Var& value)
+{
+    return CaseInsensitiveObjectWrapper {value.extract<Poco::JSON::Object::Ptr>()};
+}
```

Comment:
maybe this two function move common lib?

Reply:
Prepared PR: https://github.com/aosedge/aos_core_common_cpp/pull/24/files